library/worldgen/surface_rule/src/main/java/org/quiltmc/qsl/worldgen/surface_rule/api/SurfaceRuleEvents.java

```diff
@@ -46,6 +49,25 @@ public final class SurfaceRuleEvents {
 			callback.modifyOverworldRules(context);
 		}
 	});
+	/**
+	 * A {@link CodecMap} for the {@link OverworldModifierCallback} event. Can be used to register codecs for loading event callbacks from datapacks.
+	 */
+	public static final CodecMap<OverworldModifierCallback> MODIFY_OVERWORLD_CODECS = new CodecMap<>(context -> {});
+	/**
+	 * A {@link DynamicEventCallbackSource} for the {@link OverworldModifierCallback} event. Can be used to register
+	 * event callbacks alongside an identifier, allowing them to be overridden with datapacks.
+	 */
+	public static final DynamicEventCallbackSource<OverworldModifierCallback> MODIFY_OVERWORLD_IDENTIFIER = new DynamicEventCallbackSource<>(
```

Comment:
This field name doesnt make that much sense.

Reply:
Yeah, I was debating the name here. The idea is that it's sorta like the normal event, except callbacks have identifiers; I'm really not sure how best to name it, with that in mind.

Reply:
I think I've got a better alternative now